metrics/prometheus.go

```diff
@@ -1768,6 +1768,64 @@ func NewPrometheusCollector(i infoProvider, f ContainerLabelsFunc, includedMetri
 		})
 	}
 
+	if includedMetrics.Has(container.PSITotalMetrics) {
+		c.containerMetrics = append(c.containerMetrics, []containerMetric{
+			{
+				name:        "container_cpu_psi_total_seconds",
+				help:        "Total time spent under cpu pressure in seconds.",
+				valueType:   prometheus.CounterValue,
+				extraLabels: []string{"kind"},
+				getValues: func(s *info.ContainerStats) metricValues {
+					return getPSIValues(s, &s.Cpu.PSI, "total")
+				},
+			}, {
+				name:        "container_memory_psi_total_seconds",
+				help:        "Total container time spent under memory pressure in seconds.",
+				valueType:   prometheus.CounterValue,
+				extraLabels: []string{"kind"},
+				getValues: func(s *info.ContainerStats) metricValues {
+					return getPSIValues(s, &s.Memory.PSI, "total")
+				},
+			}, {
+				name:        "container_io_psi_total_seconds",
+				help:        "Total time spent under io pressure in seconds.",
+				valueType:   prometheus.CounterValue,
+				extraLabels: []string{"kind"},
+				getValues: func(s *info.ContainerStats) metricValues {
+					return getPSIValues(s, &s.DiskIo.PSI, "total")
+				},
+			},
+		}...)
+	}
+
+	if includedMetrics.Has(container.PSIAvgMetrics) {
+		makePSIAvgMetric := func(controller, window string) containerMetric {
+			return containerMetric{
+				name:        fmt.Sprintf("container_%s_psi_avg%s_ratio", controller, window),
+				help:        fmt.Sprintf("Ratio of time spent under %s pressure over time window of %s seconds", controller, window),
+				valueType:   prometheus.GaugeValue,
+				extraLabels: []string{"kind"},
+				getValues: func(s *info.ContainerStats) metricValues {
+					switch controller {
+					case "cpu":
+						return getPSIValues(s, &s.Cpu.PSI, "avg"+window)
+					case "memory":
+						return getPSIValues(s, &s.Memory.PSI, "avg"+window)
+					case "io":
+						return getPSIValues(s, &s.DiskIo.PSI, "avg"+window)
+					default:
+						return nil
+					}
+				},
+			}
+		}
+		for _, controller := range []string{"cpu", "memory", "io"} {
+			for _, window := range []string{"10", "60", "300"} {
+				c.containerMetrics = append(c.containerMetrics, makePSIAvgMetric(controller, window))
+			}
+		}
+	}
```

Comment:
These metrics are unnecessary in Prometheus as we can compute averages from the counters. Please remove them to avoid excess metric cardinality.

```suggestion
```

Reply:
it depends on the use case, for example psi avg10 provides new info for a regular 1m scrape interval

Reply:
I believe this may be resolved since the `10/60/300` averages are [inherent](https://github.com/rexagod/runc/blob/edd00eb3cbd223ab565c1fed975a69260be26d02/libcontainer/cgroups/stats.go#L35-L40) to the exposed PSI data?